codegen/src/impl_trait/mod.rs

```diff
@@ -1227,19 +1223,34 @@ struct ForTy {
 }
 
 impl ForTy {
-    /// Expands [`Generics`] as [`ImplGenerics`] and [`WhereClause`].
-    ///
-    /// [`ImplGenerics`]: syn::ImplGenerics
-    fn higher_rank_generics(
-        &self,
-    ) -> (Option<syn::ImplGenerics<'_>>, Option<&syn::WhereClause>) {
-        (
-            self.generics.as_ref().map(|gens| {
-                let (impl_gens, _, _) = gens.split_for_impl();
-                impl_gens
-            }),
-            self.where_clause.as_ref(),
-        )
+    /// Merges [`Generics`] of this [`ForTy`] with [`Generics`] belonging to the
+    /// trait.
+    fn merge_generics(&self, generics: &syn::Generics) -> syn::Generics {
```

Comment:
@50U10FCA7 code of this function repeats in 2 places and is not that trivial. It should be extracted and reused.